### ISSUE TITLE
Using boosted trees without an estimator

### ISSUE THREAD

**Issue by calledbymountains** [tensorflow/tensorflow#18400]
Have I written custom code : NO
OS Platform and Distribution : CentOS
TensorFlow installed from : Sources
TensorFlow version : r1.8
Bazel version : 0.11
CUDA/cuDNN version : 9.1/7.0
GPU model and memory : GTX 1080 Ti
Exact command to reproduce : N/A

At the moment is it  possible to use the TensorFlow implementation of boosted trees without invoking the concept of an Estimator ?
[Examples](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/boosted_trees) , demonstrate its usage using estimator. 

Can it be used as just another node in a TensorFlow graph ?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 18 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jart** [comment]
@yk5 Here's some feedback on boosted trees. PTAL

**Comment by yk5** [comment]
@sshrdp @nataliaponomareva please comment if not a recommended way:

Hi, for this `contrib.boosted_trees` library, tree model class is public so that you could do similar to what's seen in [`model_fn.py`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/boosted_trees/estimator_batch/model.py#L108) -- creating an object by hand and calling relevant functions - `train()` / `predict()`.

For the new boosted_trees library that's migrated into core ([link](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/estimator/canned/boosted_trees.py#L701)), we mainly provide only Estimator API but keep the model function (`_bt_model_fn`) as internal, until we're sure what'd be the best API exposed to users. Thus you could play around with the model fn out of the Estimator API we provide though, but please note that the function interface is subject to changes.

Thanks,
Younghee

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by nataliaponomareva** [comment]
In general, as yk5 stated, using an estimator is the easiest way, however nothing prevents you (at least in a contrib version) from using the methods from gbdt_batch.py (contrib version) directly. However keep in mind that many things are hidden from you - sessions, head, loss calculation, which you will have to maintain yourself if you put it as a node in a graph. Also take a look at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/boosted_trees/estimator_batch/dnn_tree_combined_estimator.py - this gives an example on how to create a graph that does a neural net and a gdbt, may be you can get some inspiration there

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
